webapp/src/pages/EditChatroomMemberPage/components/ToggleSwitchSet.tsx

```diff
@@ -0,0 +1,122 @@
+import { ToggleSwitch } from '../../../shared/components';
+import { UpdateChatroomMemberDto } from '../../../shared/generated/models/UpdateChatroomMemberDto';
+import { ResponseError } from '../../../shared/generated';
+import React, { useEffect, useState } from 'react';
+import { chatApi } from '../../../shared/services/ApiService';
+import { useNotificationContext } from '../../../shared/context/NotificationContext';
+import { ChatroomMember } from '../../../shared/generated/models/ChatroomMember';
+import { Chatroom } from '../../../shared/generated/models/Chatroom';
+import { User } from '../../../shared/generated/models/User';
+
+type CanEditParams = {
+  chatroom: Chatroom | null;
+  destCrMember: ChatroomMember | null;
+  destUser: User | null;
+  authCrMember: ChatroomMember | null;
+  authUserId: string;
+};
+
+type ToggleSwitchSetProps = {
+  isLoading: boolean;
+  chatroomId: string;
+  canEditParams: CanEditParams;
+};
+
+export function CanEdit({
+  chatroom,
+  destCrMember,
+  destUser,
+  authCrMember,
+  authUserId,
+}: CanEditParams): boolean[] {
+  const isAuthOwner = authUserId === chatroom?.ownerId ?? false;
+  const isAuthAdmin = authCrMember?.admin ?? false;
+  const isAuthBanned = authCrMember?.banned ?? true;
+  const isDestOwner = chatroom?.ownerId === destUser?.id ?? true;
+  const isDestAdmin = destCrMember?.admin ?? true;
+  return [
+    (isAuthOwner || (isAuthAdmin && !isAuthBanned && !isDestAdmin)) &&
```

Comment:
I think we should find a way to put this logic in the backend. Maybe create a table with permissions? For now, this was my answer